README.md

```diff
@@ -31,11 +31,31 @@ Allowable changes:
 - For major versions: All changes are permitted.
 - For minor versions: TBD
 
-## Naming guidelines
+## Schema Modeling Guidelines
 
-The following defines guidelines used to produce configuration schema:
+The following guidelines are used to model the configuration schema:
 
-1. To remove redundant information from the configuration file, prefixes for data produced by each of the providers
+* To remove redundant information from the configuration file, prefixes for data produced by each of the providers
 will be removed from configuration options. For example, under the `meter_provider` configuration, metric readers will be
 identified by the word `readers` rather than by `metric_readers`. Similarly, the prefix `span_` will be dropped for tracer
-provider configuration, and `logrecord` for logger provider.
+provider configuration, and `logrecord` for logger provider. 
+* Avoid use of object arrays, which are difficult to manipulate with environment variable overrides. Instead, separate the configuration of the elements from
+  their ordering. If multiple of the same element are allowed, use keys of matching `^(?<elementName>[a-zA-Z0-9_]*)(\/(?<elementId>[a-zA-Z0-9_]*))?$`. For
+  example, when configuring the span processor pipeline, use:
+   ```
+   tracer_provider:
+     processors:
+       batch/local:
+         exporter:
+           otlp:
+             endpoint: http://localhost:4317
+       batch/remote:
+         exporter:
+           otlp:
+             endpoint: http://remote-host:4317      
+     processor_pipeline: [batch/local, batch/remote]
```

Comment:
why is `processor_pipeline` needed?

Reply:
Versus just using relying on the map of `.tracer_provider.processors`? 

Order matters with processors and there's no guarantee that libraries used to parse YAML will provide access to map entries in the order they appear in the YAML. In java, I believe snakeYAML uses `LinkedHashMap` internally to represent objects so that order is retained, but there's no guarantee this is true in other implementation.

Additionally, the process pipeline gives an easy way to disable a processor via env variable without having to support some special env var syntax to "erase" an object. I.e. to remove `batch/remote`, set `OTEL_CONF_TRACER_PROVIDER__PROCESSOR_PIPELINE=batch/local`.